vassal-app/src/main/java/VASSAL/build/module/ChessClockControl.java

```diff
@@ -0,0 +1,843 @@
+/*
+ *
+ * Copyright (c) 2020 by Brian Reynolds and Michael Kiefte
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License (LGPL) as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+package VASSAL.build.module;
+
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+
+import javax.swing.JButton;
+import javax.swing.JMenuItem;
+import javax.swing.JPopupMenu;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import VASSAL.build.AbstractConfigurable;
+import VASSAL.build.AutoConfigurable;
+import VASSAL.build.Buildable;
+import VASSAL.build.GameModule;
+import VASSAL.build.module.chessclockcontrol.ChessClock;
+import VASSAL.build.module.documentation.HelpFile;
+import VASSAL.command.Command;
+import VASSAL.command.CommandEncoder;
+import VASSAL.command.NullCommand;
+import VASSAL.configure.Configurer;
+import VASSAL.configure.ConfigurerFactory;
+import VASSAL.configure.IconConfigurer;
+import VASSAL.configure.NamedHotKeyConfigurer;
+import VASSAL.configure.StringEnum;
+import VASSAL.i18n.Resources;
+import VASSAL.tools.SequenceEncoder;
+import VASSAL.tools.UniqueIdManager;
+import VASSAL.tools.LaunchButton;
+import VASSAL.tools.NamedKeyStroke;
+import VASSAL.tools.NamedKeyStrokeListener;
+import org.w3c.dom.Element;
+
+/**
+ * CHESS CLOCK CONTROLLER for VASSAL.
+ *
+ * This Module-level component manages a set of one or more {@link ChessClock} clocks. It creates a master "Chess Clocks" button on the
+ * Module's toolbar. Clicking the button shows/hides the clocks themselves, when no clocks are running, or stops all clocks if one
+ * is running. The Chess Clock Control component also contains the configurable elements that specify how the clocks are to be displayed.
+ */
+public class ChessClockControl extends AbstractConfigurable
+        implements CommandEncoder, GameComponent, UniqueIdManager.Identifyable {
+  protected static UniqueIdManager idMgr = new UniqueIdManager("ChessClockControl"); //$NON-NLS-1$
```

Comment:
Should be `final`, with an UPPERCASE_NAME

Reply:
And if the class is designed to be subclassable, also `private`, subclasses will need their own IdManager.

If the class is not designed to be subclassable, make the class `final` and replace all `protected` with `private`.

Reply:
The `idMgr`, should be `final`, and since it then would be a constant it should have an uppercase name, `ID_MGR`.

Like what we did here, but since this is new code it's not too late to change the field name to the proper convention for final static fields: https://github.com/vassalengine/vassal/pull/352/files

Reply:
done